src/Hadolint/Rule/DL3041.hs

```diff
@@ -36,7 +37,29 @@ dnfPackages args =
 
 packageVersionFixed :: Text.Text -> Bool
 packageVersionFixed package =
-  "-" `Text.isInfixOf` package || ".rpm" `Text.isSuffixOf` package
+  let parts = splitOn "-" (Text.unpack package)
+      numParts = length parts
+  in  if numParts <= 1
+      then False  -- No dashes, definitely no version
+      else if ".rpm" `Text.isSuffixOf` package
+           then True -- rpm files always have a version
+           else isVersionLike (map Text.pack $ drop 1 parts)
+
```

Comment:
Nit, just for you : You can avoid nesting conditionals by using pattern matching, e.g. like so:
```
packageVersionFixed :: Text.Text -> Bool
packageVersionFixed package
  | length parts <= 1 = False  -- No dashes, definitively no version
  | ".rpm" `Text.isSuffixOf` package = True  -- rpm files always have a version
  | otherwise = isVersionLike $ drop 1 parts
  where
    parts = Text.splitOn "-" package
```

Additionally, using `Text.splitOn` instead of `Data.List.Split.splitOn` allows you to work without repeatedly packing and unpacking the `Text` data.